File: jump-game/jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        // check every position is reachable by any of its previous positions
        for(int i = 1; i < nums.size() ;++i) {
            bool f = false;
            for(int j = i-1; j >=0; --j) {
                if (nums[j]>= i-j) {
                    f = true;
                    break;
                }
            }
            if (!f)
                return false;
        }
        return true;
    }
};

File: jump-game/jump-game.py
class Solution:

        
    def canJump(self, nums: List[int]) -> bool:
        
        for i in range(1, len(nums)):
            flag = False
            print("checking", nums[i])
            for j in range(i-1, -1, -1):
                print(i-j, ">>>",nums[j])
                if i-j<= nums[j]:
                    flag = True
                    break
            if flag == False:
                return False
            
        return True
        